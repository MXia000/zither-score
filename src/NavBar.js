import React, {Component} from 'react';
import {
    Navbar,
    Nav,
    NavDropdown,
    Form,
    FormControl,
    Button
    } from 'reactstrap';


class NavBar extends Component {
    render(){
        return (
            <div>
            </div>
        )
    }
}

export default NavBar;