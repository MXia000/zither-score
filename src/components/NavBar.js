import React, {Component} from 'react';
import {
    Navbar,
    Nav,
    NavDropdown,
    Form,
    FormControl,
    Button
    } from 'react-bootstrap';


class NavBar extends Component {
    render(){
        return (
            <div>

            </div>
        )
    }
}

export default NavBar;