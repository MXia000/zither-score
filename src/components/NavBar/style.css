.background {
    background: linear-gradient(315deg, #31caa9 30%, #54cece 70%);
}
.dropdown-toggle::after {
    display:none;
}
.navbar-default {
    background-color:teal;
    background-image: none;
    background-repeat: no-repeat;
}
.icon {
    margin-right: 10px;
}