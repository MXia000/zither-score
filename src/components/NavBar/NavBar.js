import React, { Component } from "react";
import { withRouter, Redirect } from "react-router-dom";
import SearchIcon from "@material-ui/icons/Search";
import AccountCircle from "@material-ui/icons/AccountCircle";
import LibraryMusicIcon from "@material-ui/icons/LibraryMusic";
import "./style.css";
import {
  Navbar,
  Nav,
  NavDropdown,
  Form,
  FormControl,
  Button
} from "react-bootstrap";

class NavBar extends Component {
  render() {
    console.log(this.props.location);
    if (this.props.loggedInUser && this.props.location.pathname === "/Login") {
      return <Redirect to="/" />;
    }

    return (
      <div>
        <Navbar bg="secondary" expand="lg">
          <Navbar.Brand href="/Library">Logo</Navbar.Brand>
          <Navbar.Toggle aria-controls="basic-navbar-nav" />
          <Navbar.Collapse id="basic-navbar-nav">
            <Nav className="mr-auto">
              <Nav.Link href="#">Discover</Nav.Link>
            </Nav>

            <Form inline className="mx-auto">
              <FormControl
                type="text"
                placeholder="Search"
                className="mr-sm-2"
              />
              <Button variant="outline-warning">Search</Button>
            </Form>

            {this.props.loggedInUser ? (
              <>
                <Nav className="ml-auto">
                  <Nav.Link href="/EditScore">Create</Nav.Link>
                  <Nav.Link href="#">Notification</Nav.Link>
                </Nav>

                <NavDropdown
                  title={<AccountCircle color="disabled" fontSize="large" />}
                  id="basic-nav-dropdown"
                  className="dropdown-toggle"
                >
                  <NavDropdown.Item href="#">Profile</NavDropdown.Item>
                  <NavDropdown.Item href="/Library">Library</NavDropdown.Item>
                  <NavDropdown.Item href="/Settings">Settings</NavDropdown.Item>
                  <NavDropdown.Divider />
                  <NavDropdown.Item href="#">Help</NavDropdown.Item>
                  <NavDropdown.Item href="/" onClick={this.props.onSignOut}>
                    Logout
                  </NavDropdown.Item>
                </NavDropdown>
              </>
            ) : (
              <Nav.Link href="/Login">Login</Nav.Link>
            )}
          </Navbar.Collapse>
        </Navbar>
      </div>
    );
  }
}

export default withRouter(NavBar);
