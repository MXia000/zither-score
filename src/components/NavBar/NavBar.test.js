import React from 'react';
import { shallow } from 'enzyme';
import NavBar from './NavBar'

describe("NavBar", () => {
    it('render a navbar component', () => {  
        const NavBarComp = shallow(<div><NavBar /></div>);
        expect(NavBarComp).toMatchSnapshot();
    });
});