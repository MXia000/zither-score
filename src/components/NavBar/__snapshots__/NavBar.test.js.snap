// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar render a navbar component 1`] = `ShallowWrapper {}`;

exports[`NavBar render navbar without crashing 1`] = `ShallowWrapper {}`;
